crates/types/src/journal_v2/event.rs

```diff
@@ -8,24 +8,45 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
-use crate::journal_v2::raw::{TryFromEntry, TryFromEntryError};
-use crate::journal_v2::{Entry, EntryMetadata, EntryType};
-use bytestring::ByteString;
+use crate::errors::InvocationErrorCode;
+use crate::journal_v2::raw::{RawEntry, TryFromEntry, TryFromEntryError};
+use crate::journal_v2::{CommandIndex, CommandType, Encoder, Entry, EntryMetadata, EntryType};
 use serde::{Deserialize, Serialize};
-use std::collections::HashMap;
-use strum::EnumString;
+use std::num::NonZeroU32;
 
-#[derive(Debug, Clone, PartialEq, Eq, EnumString, strum::Display, Serialize, Deserialize)]
+#[derive(
+    Debug, Copy, Clone, PartialEq, Eq, strum::Display, serde::Serialize, serde::Deserialize,
+)]
 pub enum EventType {
-    Lifecycle,
-    #[strum(default)]
-    Other(String),
+    TransientError,
+    Unknown,
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct Event {
-    pub ty: EventType,
-    pub metadata: HashMap<String, ByteString>,
+pub struct TransientErrorEvent {
+    pub error_code: InvocationErrorCode,
+    pub error_message: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub error_stacktrace: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub restate_doc_error_code: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub related_command_index: Option<CommandIndex>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub related_command_name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub related_command_type: Option<CommandType>,
+
+    // Count of the same transient error
+    pub count: NonZeroU32,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "ty")]
+pub enum Event {
+    TransientError(TransientErrorEvent),
+    /// This is used when it's not possible to parse in this Restate version the event.
+    Unknown,
```

Comment:
We could capture the unknown ordinal value here `Unknown(i32)`.

Reply:
I don't think this is needed to send it back to the UI, nor it's relevant in the state machine. If we ever need this, I add it later. This data structure can be evolved easily with additive changes because it's not storage written.